<?xml version="1.0" encoding="UTF-8"?>
<launch>
	<arg name="test" default="true"/>
	<arg name="launch_rviz" default="false" />

	<arg name="bag_filename" default="kitti_0002.bag" if="$(arg test)"/>  
	<arg name="bag_filename" default="-.bag" unless="$(arg test)"/>
	
	<arg name="rviz_name" default="rosbag.rviz" />
	<arg name="rviz_config" default="$(find  pcd_to_img)/rviz/$(arg rviz_name)" />
	<!-- Publish robot states from rosbag -->
	<arg name="play_delay" default="0.4" />
	<arg name="play_rate" default="1.0" />
	<arg name="use_sim_time" default="true" />
	
	<arg name="bag_filepath" 
		default="$(find pcd_to_img)/data/$(arg bag_filename)" /> 
	<node pkg="rosbag" type="play" name="rosbag"
		args="'$(arg bag_filepath)' 
			--delay '$(arg play_delay)' 
			--rate '$(arg play_rate)' 
			--clock -s 3 -l" >
		<remap from="/kitti/camera_color_left/image_raw" to="/camera/image_raw" if="$(arg test)"/>
		<remap from="/kitti/camera_color_left/camera_info" to="/camera/camera_info" if="$(arg test)"/>
		<!-- <remap from="/kitti/camera_color_right/image_raw" to="/camera/image_raw" if="$(arg test)"/>
		<remap from="/kitti/camera_color_right/camera_info" to="/camera/camera_info" if="$(arg test)"/> -->
		<remap from="/kitti/velo/pointcloud" to="/os_cloud_node/points" if="$(arg test)"/>
	</node>
	<!-- append - -loop for looping -->

	
	<node name="rviz_rosbag" pkg="rviz" type="rviz"
		args="-d $(arg rviz_config)"
		if="$(arg launch_rviz)">
	</node>
</launch>
